ibis/expr/types/generic.py

```diff
@@ -1808,32 +1808,20 @@ def topk(
         Table
             A top-k expression
         """
+        from ibis.expr.types.relations import bind
 
-        from ibis.expr.analysis import find_first_base_table
-
-        arg_table = find_first_base_table(self.op()).to_expr()
+        try:
+            (table,) = self.op().relations
+        except ValueError:
+            raise com.IbisTypeError("Unable to determine table for topk")
```

Comment:
The other places where you re-raise a `IbisTypeError` you raise something like

```suggestion
            raise com.IbisTypeError("TopK must depend on exactly one table.")
```

Reply:
Updated.